tests/syslog/test_syslog_source_ip.py

```diff
@@ -0,0 +1,560 @@
+                            v["syslog_server_ip"], source_ip, port, vrf))
+                    logger.info("Vrf {},{}: send syslog msg check pass".format(vrf, k))
+
+            logger.info("Create thread for {} to check syslog msg is sent".format(vrf))
+            th = threading.Thread(target=check_syslog_one_vrf, args=(routed_interfaces, port, vrf))
+            thread_pool.append(th)
+            th.start()
+
+        for thread in thread_pool:
+            thread.join(60)
+
+    def is_link_local_ip(self, ip):
+        if ip.startswith("fe80::"):
+            logger.info("link_local address {} is not supported, the relevant case should be skipped".format(ip))
+            return True
+        return False
+
+    def check_syslog_msg_is_stopped(self, routed_interfaces, mgmt_interface, port, vrf_list, is_set_source):
+        thread_pool = []
+        for vrf in vrf_list:
+            def check_no_syslog_one_vrf(routed_interfaces, port, vrf):
+                tcpdump_file = self.gen_tcpdump_cmd_and_capture_syslog_packets(routed_interfaces, port, vrf)
+                for k, v in SYSLOG_TEST_DATA[vrf].items():
+                    source_ip = v["source_ip"].split("/")[0] if is_set_source else None
+                    pytest_assert(
+                        not self.verify_syslog_packets(tcpdump_file,
+                                                       syslog_server_ip=v["syslog_server_ip"],
+                                                       port=port,
+                                                       source=source_ip),
+                        "Syslog packet with dest_ip:{}, source_ip:{}, port:{}  is not stopped on vrf of {}".format(
+                            v["syslog_server_ip"], source_ip, port, vrf))
+                    logger.info("Vrf {},{}: stop syslog msg check pass".format(vrf, k))
+
+            logger.info("Create thread for {} to check syslog msg is stopped".format(vrf))
+            th = threading.Thread(target=check_no_syslog_one_vrf, args=(routed_interfaces, port, vrf))
+            thread_pool.append(th)
+            th.start()
+
+        for thread in thread_pool:
+            thread.join(60)
+
+    def gen_tcpdump_cmd_and_capture_syslog_packets(self, routed_interfaces, port, vrf):
+        if vrf == VRF_LIST[0]:
+            tcpdump_interface = routed_interfaces[0]
+        else:
+            tcpdump_interface = vrf
+        tcpdump_cmd = "sudo timeout 30 tcpdump -i {interface} port {port} -w {dut_pcap_file}".format(
+            interface=tcpdump_interface, port=port if port else SYSLOG_DEFAULT_PORT,
+            dut_pcap_file=DUT_PCAP_FILEPATH.format(vrf=vrf))
+        tcpdump_file = capture_syslog_packets(self.duthost, tcpdump_cmd)
+        return tcpdump_file
+
+    @pytest.mark.parametrize("syslog_config_combination_case", SYSLOG_CONFIG_COMBINATION_CASE)
+    def test_basic_syslog_config(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_frontend_asic_index,
```

Comment:
@JibinBao can we have only a test cases in `TestSSIP` without auxiliary stuff?

Reply:
The reason for putting the auxiliary stuff in the TestSSIP is to make it easy to call it without pass many arguments.
In addition, the auxiliary stuff is just used for the TestSSIP.
Actually all tests are put together.